google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java

```diff
@@ -124,4 +125,19 @@ public int getSkippedResults() {
   public MoreResultsType getMoreResults() {
     return moreResults;
   }
+
+  @Override
+  public int countEntities() {
+    int count = 0;
+    Integer limit = ((KeyQuery) this.query).getLimit();
```

Comment:
Will it be KeyQuery only? Should we apply a check something like:
```java
if(!this.query instanceOf KeyQuery){
 // do something either return 0 or exception
}
```

Reply:
> Will it be KeyQuery only? Should we apply a check something like:
> 
> ```java
> if(!this.query instanceOf KeyQuery){
>  // do something either return 0 or exception
> }
> ```

there is no need for that check because its coming from context.

Reply:
ok